python/truelayer_signing/verify.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+# std imports
+import json
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, Mapping, Tuple, Optional
+
+# third party imports
+from jwt.algorithms import ECAlgorithm
+
+# local imports
+from .utils import HttpMethod, TlJwsBase, build_v2_jws_b64, decode_url_safe_base64
+
+
+class KeyFmt(Enum):
+    PEM = 0
+    JWKS = 1
+
+
+class TlVerifier(TlJwsBase):
+    """
+    Tl-Verifier
+    """
+    key_fmt: KeyFmt
+
+    def __init__(
+        self,
+        pkey: str,
+        key_fmt: KeyFmt,
+        method: HttpMethod = HttpMethod.POST,
+        path: str = "",
+        headers: Optional[Dict[str, str]] = None,
+        body: str = ""
+    ) -> None:
+        super().__init__(pkey, method, path, headers, body)
+        self.key_fmt = key_fmt
+
+    def verify(self, tl_signature: str) -> bool:
+        """
+        Verify the given `Tl-Signature`.
+        """
+        return tl_verify(VerifyArguments(
+            tl_signature,
+            self.pkey,
+            self.key_fmt,
+            self.path,
+            self.headers,
+            self.body,
+            self.http_method
+        ))
+
+
+@dataclass(frozen=True)
+class VerifyArguments:
+    tl_signature: str
+    pkey: str
+    key_fmt: KeyFmt
+    path: str
+    headers: Mapping[str, str]
+    body: str
+    method: HttpMethod
+
+
+def tl_verify(args: VerifyArguments) -> bool:
```

Comment:
I'd raise specific exception types from here, looking at the webhook example it'd be nice for users to be able to use some catch-all verification exception to return `UNAUTHORIZED` or some other specific way of handling this

Reply:
Updated to use a custom exception. Let me know what you think.